dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tool.java

```diff
@@ -478,10 +490,20 @@ public void setDefaultTestCwlParameterFile(String defaultTestCwlParameterFile) {
         getDefaultPaths().put(DescriptorLanguage.FileType.CWL_TEST_JSON, defaultTestCwlParameterFile);
     }
 
+    /**
+     * Calculates the descriptor types from the versions' source files.
+     * @return
+     */
     public List<String> calculateDescriptorType() {
-        Set<DescriptorLanguage.FileType> set = this.getWorkflowVersions().stream().flatMap(tag -> tag.getSourceFiles().stream()).map(SourceFile::getType).collect(
-                Collectors.toSet());
-        return Arrays.stream(DescriptorLanguage.values()).filter(lang -> !(lang.toString().equals("cwl") || lang.toString().equals("wdl"))).filter(lang -> set.contains(lang.getFileType()))
-                .map(lang -> lang.toString()).distinct().collect(Collectors.toList());
+        Set<DescriptorLanguage.FileType> set = this.getWorkflowVersions().stream()
+                .flatMap(tag -> tag.getSourceFiles().stream())
+                .map(SourceFile::getType)
+                .collect(Collectors.toSet());
+        return Arrays.stream(DescriptorLanguage.values())
+                .filter(lang -> !(lang.toString().equals("cwl") || lang.toString().equals("wdl")))
```

Comment:
I found it confusing too which was why I reformatted the lines :sweat_smile: 

Turns out `.filter(lang -> !(lang.toString().equals("cwl") || lang.toString().equals("wdl")))` wasn't actually filtering out cwl and wdl file types because the `DescriptorLanguage.toString()` function for `DescriptorLanguage.CWL` and `DescriptorLanguage.WDL` returns the short name `CWL` and `WDL` respectively, not the lowercase form. I removed this line entirely because I don't think we want to filter out CWL and WDL since tools can only be CWL or WDL